src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -639,18 +639,35 @@ private Expression ParseIntegerLiteral()
 		{
 			ValidateToken(TokenId.IntegerLiteral);
 			var text = _token.text;
+
+			var isUnsigned = false;
+			var isLong = false;
+			var numberEnd = text.Length - 1;
+			for (; numberEnd >= 0; numberEnd--)
```

Comment:
Why the `for` here? Just using `text[numberEnd]` is not enought? Sorry probably I missing something obvious or don't remember ;-)

Reply:
I ok get it! It is for supporting `UL` suffix or similar ... Right?

Reply:
Yes, it's because of the UL suffixes: there can be several characters to the suffix :)